packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -20,14 +27,278 @@ import {
+}
+
+interface CustomMenuProps extends MenuProps<OptionTypeBase, boolean> {
+  dropdownMenuWrapperClassName: string;
+}
+
+interface CustomOptionProps extends OptionProps<OptionTypeBase, boolean> {
+  optionWrapperClassName?: string;
+}
+
+type SelectEvent = {
+  action: "select-option" | "clear";
+  option?: DropdownOption;
+};
+
+type DropdownState = {
+  isDisabled: boolean;
+  selectProps: { withReadOnlyStyle: boolean; readOnly: boolean };
+};
+
+export interface DropdownComponentProps extends VibeComponentProps {
+  /**
+   * Custom style
+   */
+  className?: string;
+  /** ClassName to be added to dropdown option wrapper (dropdown-wrapper__option--reset) */
+  optionWrapperClassName?: string;
+  /** ClassName to be added to dropdown single value wrapper (dropdown-wrapper__single-value--reset) */
+  singleValueWrapperClassName?: string;
+  /** ClassName to be added to dropdown menu wrapper (dropdown-menu-wrapper) */
+  dropdownMenuWrapperClassName?: string;
+  /**
+   * Placeholder to show when no value was selected
+   */
+  placeholder?: string;
+  /**
+   * If set to true, dropdown will be disabled
+   */
+  disabled?: boolean;
+  /**
+   * If set to true, dropdown won't be editable
+   */
+  readOnly?: boolean;
+  /**
+   * Called when menu is opened
+   */
+  onMenuOpen?: (event: React.MouseEvent) => void;
+  /**
+   * Called when menu is closed
+   */
+  onMenuClose?: (event: React.MouseEvent) => void;
+  /**
+   * Called when key is pressed in the dropdown
+   */
+  onKeyDown?: (event: React.MouseEvent) => void;
+  /**
+   * Called when focused
+   */
+  onFocus?: (event: React.MouseEvent) => void;
+  /**
+   * Called when blurred
+   */
+  onBlur?: (event: React.MouseEvent) => void;
+  /**
+   * Called when selected value has changed
+   */
+  onChange?: (
+    option: DropdownOption | DropdownOption[],
+    event: SelectEvent | React.MouseEvent | React.KeyboardEvent
+  ) => void;
+  /**
+   * Called when the dropdown's input changes.
+   */
+  onInputChange?: (event: React.KeyboardEvent) => void;
+  /**
+   * If true, search in options will be enabled
+   */
+  searchable?: boolean;
+  /**
+   * The dropdown options
+   */
+  options?: DropdownOption[];
+  /**
+   * Text to display when there are no options
+   */
+  noOptionsMessage?: (() => string) | typeof NOOP;
+  /**
+   * If set to true, the menu will open when focused
+   */
+  openMenuOnFocus?: boolean;
+  /**
+   * If set to true, the menu will open when clicked
+   */
+  openMenuOnClick?: boolean;
+  /**
+   * If set to true, clear button will be added
+   */
+  clearable?: boolean;
+  /**
+   * custom option render function
+   */
+  optionRenderer?: (option: DropdownOption) => JSX.Element;
+  /**
+   * custom value render function
+   */
+  valueRenderer?: (...args: unknown[]) => unknown;
+  ValueRenderer?: (...args: unknown[]) => unknown;
+  /**
+   * custom menu render function
+   */
+  menuRenderer?: (...args: unknown[]) => unknown;
```

Comment:
Sounds like all of those can be `ReactElement`, but let's leave if for now like this so we won't break existing behaviors